src/Datadog.Trace/ServiceFabric/ServiceRemotingHelpers.cs

```diff
@@ -0,0 +1,215 @@
+#nullable enable
+
+using System;
+using System.Globalization;
+using System.Reflection;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.DuckTyping;
+using Datadog.Trace.Util;
+
+namespace Datadog.Trace.ServiceFabric
+{
+    internal static class ServiceRemotingHelpers
+    {
+        public const string AssemblyName = "Microsoft.ServiceFabric.Services.Remoting";
+
+        public const string ClientEventsTypeName = "Microsoft.ServiceFabric.Services.Remoting.V2.Client.ServiceRemotingClientEvents";
+
+        public const string ServiceEventsTypeName = "Microsoft.ServiceFabric.Services.Remoting.V2.Runtime.ServiceRemotingServiceEvents";
+
+        public const string SendRequestEventName = "SendRequest";
+
+        public const string ReceiveResponseEventName = "ReceiveResponse";
+
+        public const string ReceiveRequestEventName = "ReceiveRequest";
+
+        public const string SendResponseEventName = "SendResponse";
+
+        public static readonly IntegrationInfo IntegrationId = IntegrationRegistry.GetIntegrationInfo(nameof(IntegrationIds.ServiceRemoting));
+
+        private const string SpanNamePrefix = "service_remoting";
+
+        private static readonly Logging.IDatadogLogger Log = Logging.DatadogLogging.GetLoggerFor(typeof(ServiceRemotingHelpers));
+
+        private static readonly string ServiceFabricServiceName = EnvironmentHelpers.GetEnvironmentVariable(EnvironmentVariableNames.ServiceName);
+
+        private static readonly string ApplicationId = EnvironmentHelpers.GetEnvironmentVariable(EnvironmentVariableNames.ApplicationId);
+
+        private static readonly string ApplicationName = EnvironmentHelpers.GetEnvironmentVariable(EnvironmentVariableNames.ApplicationName);
+
+        private static readonly string PartitionId = EnvironmentHelpers.GetEnvironmentVariable(EnvironmentVariableNames.PartitionId);
+
+        private static readonly string NodeId = EnvironmentHelpers.GetEnvironmentVariable(EnvironmentVariableNames.NodeId);
+
+        private static readonly string NodeName = EnvironmentHelpers.GetEnvironmentVariable(EnvironmentVariableNames.NodeName);
+
+        public static bool AddEventHandler(string typeName, string eventName, EventHandler eventHandler)
+        {
+            string fullEventName = $"{typeName}.{eventName}";
+
+            try
+            {
+                Type? type = Type.GetType($"{typeName}, {AssemblyName}", throwOnError: false);
```

Comment:
Have you been able to test that the type lookup succeeds on a .NET Framework application? That's the largest piece of uncertainty I have when moving to automatic instrumentation

Reply:
After offline discussion, we decided to support this only on .NET Core / .NET 5 for now to avoid assembly loading issues. See e3e4aa47. I tested on .NET Framework after this change to make sure the code does not run.